### PR TITLE
Support custom anim prop on <Map />

### PR DESCRIPTION
**Pull request by maityarkodeb**
[Story](https://www.pivotaltracker.com/story/show/150112391)

### CODE REVIEW

File: packages/react-ui-map/src/Mapbox/index.js [hunk 1]
```diff
@@ -83,17 +90,20 @@ export default class Mapbox extends PureComponent {
...
-          theme[`Map-${color}`],
-          theme[`Map-${size}`],
-          theme.Map
-        )}
-      >
-        {map && children}
+      <div>
+        <div
+          style={{}}
+          id="map"
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/index.js]: `id` should *not* be here.

File: packages/react-ui-map/src/Mapbox/index.js [hunk 2]
```diff
@@ -83,17 +90,20 @@ export default class Mapbox extends PureComponent {
...
-          className,
-          theme[`Map-${color}`],
-          theme[`Map-${size}`],
-          theme.Map
-        )}
-      >
-        {map && children}
+      <div>
+        <div
+          style={{}}
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/index.js]: Why is this empty? It shouldn't be here if it's not set.

**Comment by tadjohnston** [comment]
I'm not sure how `packages/react-ui-map/src/Mapbox/index.js` made it through, but `index.js` should really only be used for exporting packages.  Can you please put the mapbox code inside of `packages/react-ui-map/src/Mapbox/Mapbox.js` instead? You can export it through `index.js`.

File: packages/react-ui-map/src/Mapbox/index.js (hunk 2 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/index.js]: removed.

File: packages/react-ui-map/src/Mapbox/index.js (hunk 1 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/index.js]: changed to use `ref`

File: packages/react-ui-map/src/Mapbox/Mapbox.js [hunk 3]
```diff
@@ -83,17 +89,19 @@ export default class Mapbox extends PureComponent {
...
-          className,
-          theme[`Map-${color}`],
-          theme[`Map-${size}`],
-          theme.Map
-        )}
-      >
-        {map && children}
+      <div>
+        <div
+          ref="map"
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: This is the incorrect usage of a `ref`. Please refer to the [docs for usage](https://facebook.github.io/react/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element)

File: packages/react-ui-map/src/Mapbox/Mapbox.js [hunk 4]
```diff
@@ -83,17 +89,19 @@ export default class Mapbox extends PureComponent {
...
+          className={classNames(
+            className,
+            theme[`Map-${color}`],
+            theme[`Map-${size}`],
+            theme.Map,
+          )}
+        >
+          {map && children}
+        </div>
+        <button className={theme.Button} onClick={this.handleClick}>Fit</button>
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: Please put these props on their own new lines. Also, why is this button `Fit` in this map file in the first place?  Is it supposed to be here???

File: packages/react-ui-map/src/Mapbox/Mapbox.js [hunk 5]
```diff
@@ -70,6 +70,12 @@ export default class Mapbox extends PureComponent {
       nextState.map !== this.state.map)
   }
 
+  handleClick() {
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: Is this supposed to be here like this with `fitBounds`?

File: packages/react-ui-map/src/Mapbox/Mapbox.js (hunk 4 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: working on changes.

File: packages/react-ui-map/src/Mapbox/Mapbox.js (hunk 5 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: working on moving to storybook

File: packages/react-ui-map/src/Mapbox/Mapbox.js (hunk 3 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: updated

File: packages/react-ui-map/src/Mapbox/Mapbox.js [hunk 6]
```diff
@@ -56,6 +57,11 @@ export default class Mapbox extends PureComponent {
       nextState.map !== this.state.map)
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { map } = this.state
+    fly(map, nextProps.boundingBox)
```
Comment (lfender6445) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: could we add some defaultProps for boundingBox?

Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/Mapbox.js]: I'll add the check for bounding box

**Comment by tadjohnston** [comment]
Looks great! Nice job 👍

**Comment by lfender6445** [comment]
👍

**Pull request merged**